### PR TITLE
Azure Workload Cluster Release 15.1.0

### PR DESCRIPTION
**Pull request by T-Kukawka**
https://github.com/giantswarm/giantswarm/issues/18067

### CODE REVIEW

**Comment by stone-z** [comment]
@T-Kukawka FYI the azure conformance tests aren't currently working due to changes in sonobuoy upstream. I'm currently working on fixing it but your test for now will fail

**Comment by T-Kukawka** [comment]
@stone-z isn't that the e2e test that i am running? :D i thought it is different

**Comment by stone-z** [comment]
It still uses sonobuoy to run the tests behind the scenes

**Comment by stone-z** [comment]
Actually since yours isn't pulling upstream plugins it might work. Let me undo some changes in test-infra and if it fails just re-run it

**Comment by T-Kukawka** [comment]
@stone-z no pressure, i can run it tomorrow as well as gotta continue with more tests like upgrades etc 😄

**Comment by stone-z** [comment]
TIL Tekton evaluates Tasks at execution time and not at trigger time, so you can change the tires while the car is running :rocket:

**Comment by ubergesundheit** [comment]
Unsure if related, but I've seen happa doing weird things with `15.1.0` and the `external-dns` version.

I think this because this release is missing the `componentVersion` field for `external-dns`. ([Its there for 15.0.0](https://github.com/giantswarm/releases/blob/master/azure/v15.0.0/release.yaml#L17-L19))

image

File: azure/v15.0.0/release.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ spec:
   - name: etcd
     version: 3.4.14
   date: "2021-06-14T11:40:38Z"
-  state: active
+  state: deprecated
```
Comment (whites11) [inline azure/v15.0.0/release.yaml:55]: Not sure about this. 15.1.0 is on a different minor. I would leave this as `active` TBH

File: azure/v15.1.0/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# :zap: Giant Swarm Release v15.1.0 for Azure :zap:
+
+This is the Giant Swarm Azure stable release featuring Kubernetes [1.20.8](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#v1208)
```
Comment (whites11) [inline azure/v15.1.0/README.md:3]:
```suggestion
This is the first Giant Swarm Azure stable release featuring Kubernetes [1.20.8](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#v1208).
```

File: azure/v15.1.0/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Release introduces feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/)
```
Comment (whites11) [inline azure/v15.1.0/README.md:5]:
```suggestion
Release introduces feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/).
```

File: azure/v15.1.0/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+- Update the Debian images to pick up CVE fixes in the base images:
+  - Update the `debian-base` image to v1.7.0
+  - Update the `debian-iptables` image to v1.6.1 ([#102341](https://github.com/kubernetes/kubernetes/pull/102341), [@cpanato](https://github.com/cpanato)) [SIG API Machinery and Testing]_
+#### Changed
+- sigs.k8s.io/apiserver-network-proxy/konnectivity-client: v0.0.15 → v0.0.19
+
+
+### containerlinux [2765.2.6](https://www.flatcar-linux.org/releases/#release-2765.2.6)
+
+#### Security fixe
```
Comment (whites11) [inline azure/v15.1.0/README.md:56]:
```suggestion
#### Security fixes
```

File: azure/v15.1.0/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+- Linux ([CVE-2020-26558](https://nvd.nist.gov/vuln/detail/CVE-2020-26558), [CVE-2021-0129](https://nvd.nist.gov/vuln/detail/CVE-2021-0129), [CVE-2020-24587](https://nvd.nist.gov/vuln/detail/CVE-2020-24587), [CVE-2020-24586](https://nvd.nist.gov/vuln/detail/CVE-2020-24586), [CVE-2020-24588](https://nvd.nist.gov/vuln/detail/CVE-2020-24588), [CVE-2020-26139](https://nvd.nist.gov/vuln/detail/CVE-2020-26139), [CVE-2020-26145](https://nvd.nist.gov/vuln/detail/CVE-2020-26145), [CVE-2020-26147](https://nvd.nist.gov/vuln/detail/CVE-2020-26147), [CVE-2020-26141](https://nvd.nist.gov/vuln/detail/CVE-2020-26141), [CVE-2021-3564](https://nvd.nist.gov/vuln/detail/CVE-2021-3564), [CVE-2021-28691](https://nvd.nist.gov/vuln/detail/CVE-2021-28691), [CVE-2021-3587](https://nvd.nist.gov/vuln/detail/CVE-2021-3587), [CVE-2021-3573](https://nvd.nist.gov/vuln/detail/CVE-2021-3573))
+#### Bug fixes**
```
Comment (whites11) [inline azure/v15.1.0/README.md:58]:
```suggestion
#### Bug fixes
```

File: azure/v15.1.0/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+- Update-engine sent empty requests when restarted before a pending reboot ([Flatcar#388](https://github.com/kinvolk/Flatcar/issues/388))
+- motd login prompt list of failed services: The output of "systemctl list-units --state=failed --no-legend" contains a bullet point which is not expected and ended up being taken as the unit name of failed units which was previously on the start of the line. Filtered the bullet point out to stay compatible with the old behavior in case upstream would remove the bullet point again. ([coreos-overlay#1042](https://github.com/kinvolk/coreos-overlay/pull/1042))
+#### Updates**
```
Comment (whites11) [inline azure/v15.1.0/README.md:61]:
```suggestion
#### Updates
```

File: azure/v15.0.0/release.yaml (hunk 1 above)
Comment (T-Kukawka) [reply azure/v15.0.0/release.yaml:55]: all right 👍

Comment (T-Kukawka) [reply azure/v15.0.0/release.yaml]: fixed

**Review by whites11 — APPROVED** [review]
GO GO GO

File: azure/v15.1.0/README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
+# :zap: Giant Swarm Release v15.1.0 for Azure :zap:
+
+This is the first Giant Swarm Azure stable release featuring Kubernetes [1.20.8](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#v1208).
+
+Release introduces feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/).
```
Comment (fiunchinho) [inline azure/v15.1.0/README.md:5]: `Release introduces feature` sounds weird to me. Maybe `This release introduces a new feature allowing...`?

File: azure/v15.1.0/README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+## Change details
+
+
+### azure-operator [5.8.0](https://github.com/giantswarm/azure-operator/releases/tag/v5.8.0)
+
+#### Added
+- Allow using an existing public IP for the NAT gateway of worker nodes.
+#### Fixed
+- Fix udev rules that caused `/boot` automount to fail
```
Comment (fiunchinho) [inline azure/v15.1.0/README.md:16]:
```suggestion
- Fix udev rules that caused `/boot` automount to fail.
```

File: azure/v15.1.0/README.md (hunk 7 above)
Comment (T-Kukawka) [reply azure/v15.1.0/README.md]: fixed

File: azure/v15.1.0/README.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 This is the first Giant Swarm Azure stable release featuring Kubernetes [1.20.8](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#v1208).
 
-Release introduces feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/).
+This release introduces a new feature feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/).
```
Comment (fiunchinho) [inline azure/v15.1.0/README.md:5]:
```suggestion
This release introduces a new feature allowing configuration of the public IP address for the NAT gateway of worker nodes for existing and new clusters. For more details please talk to your Account Engineer or follow [Giant Swarm Documentation](https://docs.giantswarm.io/advanced/egress-ip-address-azure/).
```

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**